injector/injector.go

```diff
@@ -201,67 +152,183 @@ func (ti *TrackerInjector) InjectTracker(vastXML string, NURL string) string {
 				outputXML.WriteString("]]>")
 				continue
 			}
+		default:
+			encoder.EncodeToken(t)
 		}
-
-		encoder.EncodeToken(t)
 	}
 
 	encoder.Flush()
 
-	if !inlineWrapperTagFound {
+	if !st.inlineWrapperTagFound {
 		// 	// Todo log adapter.<bidder-name>.requests.badserverresponse metrics
 		return vastXML
 	}
 	return outputXML.String()
 }
 
+func (ti *TrackerInjector) handleStartElement(tt xml.StartElement, st *InjectionState, outputXML *strings.Builder, encoder *xml.Encoder) {
+	switch tt.Name.Local {
+	case wrapperCase:
+		st.wrapperTagFound = true
+		encoder.EncodeToken(tt)
+	case creativeCase:
+		st.isCreative = true
+		for _, attr := range tt.Attr {
+			if strings.ToLower(attr.Name.Local) == "adid" {
+				st.creativeId = attr.Value
+			}
+		}
+		encoder.EncodeToken(tt)
+	case linearCase:
+		st.injectVideoClicks = true
+		st.injectTracker = true
+		encoder.EncodeToken(tt)
+	case videoClicksCase:
+		st.injectVideoClicks = false
+		encoder.Flush()
```

Comment:
For my own understanding, should we be handling the error that's returned from .Flush()?

Reply:
Flush() will only return error if the buffer capacity is less than bytes to be written. As we using string.Builder, buffer capacity will grow whenever needed.